### PR TITLE
removed return unreachables added fromPublicKey and more

### PR DESCRIPTION
**Pull request by HansWestman74**
This pull request 

- removes incorrect code that causes errors when upgrading to the latest AssemblyScript compiler. Specifically, AssemblyScript does not have a never return type like Rust and TypeScript. For this reason the latest asc compiler gets confused with code like

       return <T>unreachable().
       
- adds missing function  fromPublicKey needed for the AssemblyScript implementation of the keys-manager.
- declares and makes use of casper-blake2b in externals.ts

**bytesrepr.ts**
let bytes = bytes.subarray(currentPos); --> bytes = bytes.subarray(currentPos);
return <T>unreachable(); ---> unreachable();

**index.ts**
replaced all  return <T>unreachable() with unreachable()
getNamedArgSize(name: String) ---> getNamedArgSize(name: string)
Replaced all U32 with u32 in getNamedArgSize

**key.ts**
replaced all  return <T>unreachable() ---> unreachable()
fixed ending of Key.toBytes so that the compiler is happy with the return type
Implemented   fromPublicKey(publicKey:PublicKey) : AccountHash  in AccountHash class.

**public_key.ts**
Implemented getRawBytes and getAlgorithmName in PublicKey class

**purse.ts** 
replaced all  return <T>unreachable() ---> unreachable()

**externals.ts**
added declaration of casper_blake2b function used in implemented fromPublicKey function

### CODE REVIEW

File: smart_contracts/contract_as/assembly/key.ts [hunk 1]
```diff
@@ -4,10 +4,12 @@ import {UREF_SERIALIZED_LENGTH} from "./constants";
 import {URef} from "./uref";
 import {CLValue} from "./clvalue";
 import {Error, ErrorCode} from "./error";
-import {checkTypedArrayEqual, typedToArray} from "./utils";
+import {checkTypedArrayEqual, typedToArray,encodeUTF8} from "./utils";
```
Comment (mpapierski) [inline smart_contracts/contract_as/assembly/key.ts]:
```suggestion
import {checkTypedArrayEqual, typedToArray, encodeUTF8} from "./utils";
```

**Review by mpapierski — APPROVED** [review]
I'd suggest consistent formatting in the rest of new code but other than that 👍

**Review by Fraser999 — APPROVED** [review]
LGTM

**Comment by HansWestman74** [comment]
bors merge

**Pull request merged**